tests/jax/test_distributed_layernorm_mlp.py

```diff
@@ -0,0 +1,266 @@
+# Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+import pytest
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+from typing import Callable, Sequence, Union
+
+from jax.sharding import Mesh, NamedSharding, PartitionSpec
+""" from distributed_test_base import compare_ops """
+from transformer_engine.jax.fp8 import FP8MetaPackage, FP8Helper
+from transformer_engine.jax.fp8 import is_fp8_available
+from transformer_engine.jax import fp8_autocast
+from transformer_engine.jax.flax import LayerNormMLP
+from transformer_engine.jax.mlp import fused_layernorm_fp8_mlp
+from transformer_engine.jax.sharding import HIDDEN_AXES, HIDDEN_TP_AXES, \
+    BATCH_AXES, SEQLEN_TP_AXES, SEQLEN_AXES, \
+    W_NO_SHARD_AXES, W_FSDP_AXES, W_TP_AXES, W_JOINED_AXES
+from transformer_engine.jax.sharding import MeshResource
+from utils import assert_allclose, is_devices_enough
+
+
+is_fp8_supported, reason = is_fp8_available()
+DTYPES = [jnp.bfloat16, jnp.float16]
+INPUT_SHAPE = [[64, 64, 32]]    # [seqlen, batch, hidden_in]
```

Comment:
Let's use a shape with batch != seqlen, which can increase the coverage. Otherwise, we may miss some corner case for something related to`transpose_bs` .